pymatgen/io/common.py

```diff
@@ -0,0 +1,487 @@
+        an extremely computationally intensive process, depending on how many
+        grid points are in the VolumetricData.
+
+        Args:
+            ind (int): Index of atom.
+            radius (float): Radius of integration.
+            nbins (int): Number of bins. Defaults to 1. This allows one to
+                obtain the charge integration up to a list of the cumulative
+                charge integration values for radii for [radius/nbins,
+                2 * radius/nbins, ....].
+
+        Returns:
+            Differential integrated charge as a np array of [[radius, value],
+            ...]. Format is for ease of plotting. E.g., plt.plot(data[:,0],
+            data[:,1])
+        """
+        # For non-spin-polarized runs, this is zero by definition.
+        if not self.is_spin_polarized:
+            radii = [radius / nbins * (i + 1) for i in range(nbins)]
+            data = np.zeros((nbins, 2))
+            data[:, 0] = radii
+            return data
+
+        struct = self.structure
+        a = self.dim
+        if ind not in self._distance_matrix or self._distance_matrix[ind]["max_radius"] < radius:
+            coords = []
+            for x, y, z in itertools.product(*(list(range(i)) for i in a)):
+                coords.append([x / a[0], y / a[1], z / a[2]])
+            sites_dist = struct.lattice.get_points_in_sphere(coords, struct[ind].coords, radius)
+            self._distance_matrix[ind] = {
+                "max_radius": radius,
+                "data": np.array(sites_dist),
+            }
+
+        data = self._distance_matrix[ind]["data"]
+
+        # Use boolean indexing to find all charges within the desired distance.
+        inds = data[:, 1] <= radius
+        dists = data[inds, 1]
+        data_inds = np.rint(np.mod(list(data[inds, 0]), 1) * np.tile(a, (len(dists), 1))).astype(int)
+        vals = [self.data["diff"][x, y, z] for x, y, z in data_inds]
+
+        hist, edges = np.histogram(dists, bins=nbins, range=[0, radius], weights=vals)
+        data = np.zeros((nbins, 2))
+        data[:, 0] = edges[1:]
+        data[:, 1] = [sum(hist[0 : i + 1]) / self.ngridpts for i in range(nbins)]
+        return data
+
+    def get_average_along_axis(self, ind):
+        """
+        Get the averaged total of the volumetric data a certain axis direction.
+        For example, useful for visualizing Hartree Potentials from a LOCPOT
+        file.
+
+        Args:
+            ind (int): Index of axis.
+
+        Returns:
+            Average total along axis
+        """
+        m = self.data["total"]
+        ng = self.dim
+        if ind == 0:
+            total = np.sum(np.sum(m, axis=1), 1)
+        elif ind == 1:
+            total = np.sum(np.sum(m, axis=0), 1)
+        else:
+            total = np.sum(np.sum(m, axis=0), 0)
+        return total / ng[(ind + 1) % 3] / ng[(ind + 2) % 3]
+
+    def mask_sphere(self, radius, cx, cy, cz):
```

Comment:
I think it does because the lattice vectors do not come into play--It's just indexing the volumetric array. i.e. the cube data is stored as a cube, and then the lattice vectors let you transform that data to real space, where it's orthogonal. Does that seem right or am I off my rocker?

Reply:
So I'm not sure how volumemetric data is handled in CPMD but in VASP and by extension the VolumetricData object here, the data is stored on a grid that is regularly spaced in the direction of the lattice vectors so that definitely causes problems.

Reply:
so from what I can tell, the file is called CUBE because the name is from Gaussian which does actually use a cubic grid since they obviously don't care. 
CPMD is using the CUBE format but has the info from lattice parameters included also.
So I would bet it's the same as VASP but you would have check the source code of `cpmd2cube.x` to be 100% sure